### PR TITLE
Create room in firebase

### PR DESCRIPTION
**Pull request by DanielMuckelbauer**

### CODE REVIEW

File: src/room-creation/RoomCreationModal.svelte [hunk 1]
```diff
@@ -1,60 +1,54 @@
...
+        addRoomToFirebase(roomForm);
+        closeModal();
+    }
 
-        if (res.ok) {
-            return roomForm;
-        } else {
-            throw new Error();
-        }
+    function isIncomplete(roomForm: Room): boolean {
```
[the remark this answers is not in the document]
Comment (DanielMuckelbauer) [reply src/room-creation/RoomCreationModal.svelte:25]: You are so right, wounded by my own weapon

[the remark this answers is not in the document]
Comment (Hakem22) [reply src/room-creation/RoomCreationModal.svelte:25]: I still remember the call fight that we had once I about that hihi

[the remark this answers is not in the document]
Comment (DanielMuckelbauer) [reply src/room-creation/RoomCreationModal.svelte:25]: Well, now I can reveal it: this whole project was just a test to see if you remember the lesson! you passed with flying colors. 
I will proceed to delete the repository and google cloud project if you don't mind

File: src/firebase/add-room-to-firebase.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import type { Room } from '../room-creation/models/room';
+import { db } from './firebase';
+import { addDoc, collection } from 'firebase/firestore/lite';
+
+export function addRoomToFirebase(room: Room): void {
```
[the remark this answers is not in the document]
Comment (DanielMuckelbauer) [reply src/firebase/add-room-to-firebase.ts]: Changed it to persist at least

File: src/firebase/models/firebase-config.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface FirebaseConfig {
```
[the remark this answers is not in the document]
Comment (DanielMuckelbauer) [reply src/firebase/models/firebase-config.ts]: Done

File: src/room-creation/RoomCreationModal.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielMuckelbauer) [reply src/room-creation/RoomCreationModal.svelte]: renamed it

File: src/room-starting/Room.svelte [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
+    import { Room } from '../room-creation/models/room';
+
     export let id: string;
     export let location;
-    let room = {};
-    
-    function start() {
+    let room = {} as Room;
+
+    function start(): void {
```
[the remark this answers is not in the document]
Comment (DanielMuckelbauer) [reply src/room-starting/Room.svelte]: Lint rule is the way. So much to do, `we will easily make it till wednesday`"™

File: tests/room-creation-modal.test.ts [hunk 5]
```diff
@@ -1,13 +1,14 @@
-import { fireEvent, render } from "@testing-library/svelte";
+import { fireEvent, render, RenderResult } from '@testing-library/svelte';
 import RoomCreationModal from "../src/room-creation/RoomCreationModal.svelte";
 
 describe("Room Creation Modal", () => {
+    let component: RenderResult<RoomCreationModal>;
```
Comment (Hakem22) [inline tests/room-creation-modal.test.ts:5]: Very nice :)

**Review by Hakem22 — APPROVED** [review]

**Pull request merged**